src/safety_report_generator.py

```diff
@@ -0,0 +1,139 @@
+from invoke.context import Context
+from datetime import datetime
+
+import json
+import os
+
+TO_BE_DECIDED="To Be Decided"
+
+class SafetyReportGenerator:
+    """
+    The SafetyReportGenerator class deals with the functionality of generating safety reports for running containers.
+    The safety report takes the following format:
+    [
+        {
+            "package": "package",
+            "scan_status": "SUCCEEDED/FAILED/IGNORED",
+            "installed": "version",
+            "vulnerabilities": [
+                {
+                    "vid": "safety_vulnerability_id",
+                    "advisory": "description of the issue",
+                    "reason_to_ignore":"reason to ignore the vid",
+                    "spec": "version_spec"
+                },
+                ...
+            ]
+            "date":
+        }
+        ...
+    ]
+    """
+
+    def __init__(self, container_id, ignore_dict = {}):
+        self.container_id = container_id
+        self.vulns_dict = {}
+        self.vulns_list = []
+        self.ignore_dict = ignore_dict
+        self.ignored_vulnerability_count = {}
+        self.ctx = Context()
+        self.docker_exec_cmd = f"docker exec -i {container_id}"
+        self.safety_check_output = None
+    
+    def insert_vulnerabilites_into_report(self, vulns):
```

Comment:
Please use `vulnerabilities` or whole words instead of shortened words for variable names for clarity.

Also, please add docstrings to all functions in this class.

Reply:
Done.